src/main/java/com/amazonaws/secretsmanager/sql/AWSSecretsManagerDriver.java

```diff
@@ -354,16 +354,29 @@ public Connection connect(String url, Properties info) throws SQLException {
             return null;
         }
 
-        String unwrappedUrl = "";
         if (url.startsWith(SCHEME)) { // If this is a URL in the correct scheme, unwrap it
-            unwrappedUrl = unwrapUrl(url);
+            String unwrappedUrl = unwrapUrl(url);
+
+            if (info != null && info.getProperty("user") != null) {
+                String credentialsSecretId = info.getProperty("user");
+                try {
+                    return connectWithSecret(unwrappedUrl, info, credentialsSecretId);
+                } catch (InterruptedException e) {
+                    // User driven exception. Throw a runtime exception.
+                    throw new RuntimeException(e);
+                }
+            } else {
+                return getWrappedDriver().connect(unwrappedUrl, info);
+            }
```

Comment:
This is existing code that was previously after this `if`. It only applies if you are using a full jdbc url with the scheme, so I've moved it up here.